File: 0487-max-consecutive-ones-ii/0487-max-consecutive-ones-ii.cpp
class Solution {
public:
    int findMaxConsecutiveOnes(vector<int>& nums) {
        int answer = 0; 
        int start = 0, end = 0;
        int k = 1; 
        while(end < nums.size()){
            k -= nums[end] == 0 ? 1 : 0; 
            while(k < 0){
                k += nums[start] == 0 ? 1 : 0; 
                start++; 
            }
            answer = max(answer, end - start + 1);
            end++; 
        }   
        return answer; 
    }
};

File: 0487-max-consecutive-ones-ii/0487-max-consecutive-ones-ii.java
class Solution {
    public int findMaxConsecutiveOnes(int[] nums) {
        int answer = 0; 
        Queue<Integer> q = new ArrayDeque<>(); 
        int index = 0; 
        int start = 0; 
        int k = 1; 
        while(index < nums.length){
            
            if(nums[index] == 0){
                q.add(index); 
            }

            while(q.size() > k){
                start = q.poll() + 1; 
            }

            answer = Math.max(answer, index - start + 1); 
            index++; 
        }
        return answer; 
    }
}